src/algorithms/generate.rs

```diff
@@ -78,8 +76,9 @@ pub(crate) fn generate_multi_prime_key_with_exp<R: CryptoRngCore + ?Sized>(
         }
 
         for (i, prime) in primes.iter_mut().enumerate() {
-            *prime = rng.gen_prime(todo / (nprimes - i));
-            todo -= prime.bits();
+            let bits = (todo / (nprimes - i)) as u32;
+            *prime = generate_prime_with_rng(rng, bits, bits);
```

Comment:
```suggestion
            let bits = (todo / (nprimes - i)) as u32;
            let bits_precision = BoxedUint::zero_with_precision(bits).bits_precision();
            *prime = generate_prime_with_rng(rng, bits, bits);
```

One way to get it :woman_shrugging:

Reply:
applied manually